RoR Capstone: BudgetApp

mateo951 opened the pull request:
- Use ruby gems as software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Use params from browser request in a safe way.
- Use preprocessed html file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a webapp that requires the user to log in.
- Use devise gem for authentication.
- Limit access to webapp resources based on authorization rules.
- Analyze in writing why you have made a coding choice using one structure over another.

[VIdeo Presentation](https://www.loom.com/share/2764cce433ec4ff49df2da19d96fb7fb)

[Live App](https://budgetappmv.herokuapp.com)

Thanks in advance for your review ✔

mateo951 committed:
Add rails setup with postgreSQL DB

mateo951 committed:
Add linters

mateo951 committed:
Add hotfix for db

mateo951 committed:
Add User model, controller and views

mateo951 committed:
Add authentication functionality

mateo951 committed:
Add Category model, controllers and views

mateo951 committed:
Add Activities  model, controllers and views

mateo951 committed:
Add associations/validations for User

mateo951 committed:
Add associations/validations for Category

mateo951 committed:
Add associations/validations for Activity

mateo951 committed:
Add fixes for linters

mateo951 committed:
Merge pull request #1 from mateo951/feature/setup

BudgetApp: Feature/setup

mateo951 committed:
Add config for splash view

mateo951 committed:
Add splash view

mateo951 committed:
Add login view

mateo951 committed:
Add signup view

mateo951 committed:
Add hotfix for devise redirection

mateo951 committed:
Add hotfix Category view

mateo951 committed:
Add fix for categories/activities models & category view

mateo951 committed:
Add view for new/edit category

mateo951 committed:
Add fix for splash screen

mateo951 committed:
Add fix for splash screen v2

mateo951 committed:
Add hotfix for forms & devise registration/login

mateo951 committed:
Add hotfix for categories & Modify css of views

mateo951 committed:
Add changes to schema

mateo951 committed:
Add fix for models & validations

mateo951 committed:
Add fix for models & validations v2

mateo951 committed:
Modify categories controller to improve performance & Add more styling

mateo951 committed:
Add Activity views and controllers & Add views

mateo951 committed:
Add README & linters

mateo951 committed:
Add fix for linters v1

mateo951 committed:
Add fix for linters v2

mateo951 committed:
Merge pull request #2 from mateo951/feature/views

Feature/views

mateo951 committed:
Add fix for heroku v1

mateo951 committed:
Add fix for heroku v1

mateo951 committed:
Add fix for heroku v2

mateo951 committed:
Add fix for heroku v3

mateo951 committed:
Add fix for heroku v4

mateo951 committed:
Add fix for heroku v5

mateo951 committed:
Update README.md

mateo951 committed:
Update README.md

Davidosky007 reviewed (changes requested):
Hi @mateo951 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Nice video presentation 👍 
* No linter errors 👍 
* Well written readme file 👍 
* Nice PR description 👍 

## Required Changes ♻️
* Kindly Create unit and integration tests for all the most important components of your RoR application.
* Kindly ensure The project is deployed and accessible online as the link provided is broken kindly fix it .
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mateo951 committed:
Add unit testing

mateo951 committed:
Add fix for linters

mateo951 committed:
Add fix for linters v2

mateo951 committed:
Add fix for linters v3

mateo951 committed:
Add fix for linters v4

mateo951 committed:
Add fix for linters v5

mateo951 committed:
Add fix for linters v6

mateo951 committed:
Add fix for linters v7

mateo951 committed:
Merge pull request #4 from mateo951/feature/testing

BudgetApp: Add unit testing

mateo951 commented:
- Added unit testing 
   
![image](https://user-images.githubusercontent.com/15807343/178128509-801e5f08-bc65-4a0b-bcfe-ac54ec268e1d.png)

- Fixed [Heroku](https://budgetappmv.herokuapp.com) deployment 🤓 (finally) 
![image](https://user-images.githubusercontent.com/15807343/178128491-a621386f-831d-4973-b42f-ddbd8936dd9e.png)

Cathella reviewed (approved):
## 🔥 Status: Approved 🟢

### Hi @mateo951 👋🏻, 

Great job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

![rosycheeks-mochi-peach](https://user-images.githubusercontent.com/4247740/178129097-756acb44-b14c-4321-9735-93753fe34f0e.gif)


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
